### PR TITLE
API: integer values and non-nano dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
I'm open to suggestions for a better wording on the whatsnew.

Note this handles all-int cases but not mixed-int.

timedelta64 analogue coming up.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
test failures on circleci look related?

**Comment by jbrockmendel** [comment]
> test failures on circleci look related?

yep, and there was even a helpful comment that telling me not to do what i did.  hopefully fixed now

**Review by MarcoGorelli — APPROVED** [review]
Nice! lgtm

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -628,6 +628,7 @@ Other API changes
 - Passing ``dtype`` of "timedelta64[s]", "timedelta64[ms]", or "timedelta64[us]" to :class:`TimedeltaIndex`, :class:`Series`, or :class:`DataFrame` constructors will now retain that dtype instead of casting to "timedelta64[ns]"; passing a dtype with lower resolution for :class:`Series` or :class:`DataFrame` will be cast to the lowest supported resolution "timedelta64[s]" (:issue:`49014`)
 - Passing a ``np.datetime64`` object with non-nanosecond resolution to :class:`Timestamp` will retain the input resolution if it is "s", "ms", "us", or "ns"; otherwise it will be cast to the closest supported resolution (:issue:`49008`)
 - Passing ``datetime64`` values with resolution other than nanosecond to :func:`to_datetime` will retain the input resolution if it is "s", "ms", "us", or "ns"; otherwise it will be cast to the closest supported resolution (:issue:`50369`)
+- Passing integer values and a non-nanosecond datetime64 dtype (e.g. "datetime64[s]") :class:`DataFrame`, :class:`Series`, or :class:`Index` will treat the values as multiples of the dtype's unit, matching the behavior of e.g. ``Series(np.array(values, dtype="M8[s]"))`` (:issue:`??`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:631]:
```suggestion
- Passing integer values and a non-nanosecond datetime64 dtype (e.g. "datetime64[s]") :class:`DataFrame`, :class:`Series`, or :class:`Index` will treat the values as multiples of the dtype's unit, matching the behavior of e.g. ``Series(np.array(values, dtype="M8[s]"))`` (:issue:`51092`)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**